src/NextGen/DonationForm/Repositories/DonationFormRepository.php

```diff
@@ -28,161 +25,262 @@ class DonationFormRepository {
+     * @param DonationForm $donationForm
+     *
+     * @return void
+     * @throws Exception|InvalidArgumentException
      */
-    public function getFormDataGateways(int $formId): array
+    public function update(DonationForm $donationForm)
     {
-        $formDataGateways = [];
+        $this->validateProperties($donationForm);
 
-        foreach ($this->getEnabledPaymentGateways($formId) as $gateway) {
-            $gatewayId = $gateway->getId();
+        Hooks::doAction('givewp_donation_form_updating', $donationForm);
 
-            $formDataGateways[$gatewayId] = array_merge(
-                [
-                    'label' => give_get_gateway_checkout_label($gatewayId) ?? $gateway->getPaymentMethodLabel(),
-                ],
-                method_exists($gateway, 'formSettings') ? $gateway->formSettings($formId) : []
-            );
+        $date = Temporal::getCurrentFormattedDateForDatabase();
+
+        DB::query('START TRANSACTION');
+
+        try {
+            DB::table('posts')
+                ->where('ID', $donationForm->id)
+                ->update([
+                    'post_modified' => $date,
+                    'post_modified_gmt' => get_gmt_from_date($date),
+                    'post_status' => $donationForm->status->getValue(),
+                    'post_title' => $donationForm->formTitle,
+                    'post_content' => $donationForm->blocksContent,
+                ]);
+
+            DB::table('give_formmeta')
+                ->where('form_id', $donationForm->id)
+                ->where('meta_key', DonationFormMetaKeys::SETTINGS()->getValue())
+                ->update([
+                    'meta_value' => json_encode($donationForm->settings),
+                ]);
+
+        } catch (Exception $exception) {
+            DB::query('ROLLBACK');
+
+            Log::error('Failed updating a donation form', compact('donationForm'));
+
+            throw new $exception('Failed updating a donation form');
         }
 
-        return $formDataGateways;
-    }
+        DB::query('COMMIT');
 
+        Hooks::doAction('givewp_donation_form_updated', $donationForm);
+    }
 
     /**
      * @unreleased
      *
-     * @throws EmptyNameException|TypeNotSupported
+     * @param DonationForm $donationForm
+     *
+     * @throws Exception
      */
-    public function createFieldsApiForm(int $formId): Form
+    public function delete(DonationForm $donationForm): bool
     {
-        $donationForm = new Form($formId);
+        DB::query('START TRANSACTION');
 
-        $formBlockData = json_decode(get_post($formId)->post_content, false);
+        Hooks::doAction('givewp_donation_form_deleting', $donationForm);
 
-        foreach ($formBlockData as $block) {
-            $donationForm->append($this->convertFormBlocksToSection($block));
-        }
+        try {
+            DB::table('posts')
+                ->where('id', $donationForm->id)
+                ->delete();
+        } catch (Exception $exception) {
```

Comment:
We'll also need to delete the formMeta (settings)

Reply:
Resolved by https://github.com/impress-org/givewp-next-gen/pull/63/commits/5122a3ce7bfe4044bab8c5b7ac1c3cd914df7340.